x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexOperator.java

```diff
@@ -37,47 +39,64 @@
 
 // TODO rename package
 public final class LookupFromIndexOperator extends AsyncOperator<LookupFromIndexOperator.OngoingJoin> {
+    public record MatchConfig(FieldAttribute.FieldName fieldName, int channel, DataType type) implements Writeable {
+        public MatchConfig(FieldAttribute match, Layout.ChannelAndType input) {
+            // TODO: Using exactAttribute was supposed to handle TEXT fields with KEYWORD subfields - but we don't allow these in lookup
+            // indices, so the call to exactAttribute looks redundant now.
+            this(match.exactAttribute().fieldName(), input.channel(), input.type());
+        }
+
+        public MatchConfig(StreamInput in) throws IOException {
+            this(new FieldAttribute.FieldName(in.readString()), in.readInt(), DataType.readFrom(in));
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(fieldName.string());
+            out.writeInt(channel);
+            type.writeTo(out);
+        }
```

Comment:
It's fairly surprising to see something being `Serializable` here.

Reply:
Could you make this a top level class and add an `AbstractWireSerializingTestCase` for it? It's a nice layer of paranoia to make sure we don't typo something.